scripts/zones/Horlais_Peak/mobs/Fighting_Sheep.lua

```diff
@@ -12,6 +12,10 @@ entity.onMobInitialize = function(mob)
     mob:setMod(xi.mod.ICE_MEVA, 75) -- Todo: Move to mob_resists.sql
 end
 
+entity.onMobSpawn = function(mob)
+    mob:addMod(xi.mod.REGAIN, 3000)
```

Comment:
Another way to go about this which I believe would be more accurate would be to replace their auto attacks with this ability. You can do this by passing `mob:SetMobSkillAttack(####)` with the number being the skill list ID.

Reply:
oooooo I like that. I'll get on it when I can

Reply:
Note you'll need two skill lists because of this, one for their autoskill-as-melee, and one for their other skills as they actually can use sheep song if they get sufficient TP to do so through being attacked. Probably lamb chop and rage as well but I haven't personally seen it. Unlike most autoskills, SE didn't suppress the "uses" message on these guys so you don't have to create a duplicate skill at least (different params would have been needed to suppress the message).

Reply:
As far as I can tell from the capture in discord they don't use any other skills. But in other youtube videos I can find they use all other sheep skills minus regular sheep charge.

Reply:
ah nevermind! they seem to also use sheep song.

Reply:
> As far as I can tell from the capture in discord they don't use any other skills. 

Captures will only pick up on what happened at the time a capture was being done. This is why we have to do so much capturing to get everything.

> But in other youtube videos I can find they use all other sheep skills minus regular sheep charge.

:+1: 

> ah nevermind! they seem to also use sheep song.

I'd personally seen sheep song so I knew that one, just didn't know for sure on the others :)

Reply:
One issue that comes in with this is that even though its an auto attack, replacing it with a skill uses up the mob's TP move every time they auto attack. Encountered this with Geush Urvan never using TP moves because of me doing the same thing. I had to do this to get him to be able to use TP moves: 

```
entity.onMobWeaponSkillPrepare = function(mob, target)
    mob:setLocalVar("skill_tp", mob:getTP())
end

entity.onMobWeaponSkill = function(target, mob, skill)
    if skill:getID() == ### then 
        mob:addTP(mob:getLocalVar("skill_tp"))
        mob:setLocalVar("skill_tp", 0)
    end
end
```

Reply:
Thanks, crit! I'll put this in asap

Reply:
> One issue that comes in with this is that even though its an auto attack, replacing it with a skill uses up the mob's TP every time they auto attack. 

Thought someone pr'd to fix that bug. if not it needs fixed instead of worked around, I'll try and make it higher priority so contributors don't have to worry about it.

Reply:
according to [this comment](url) setting teh "special" value in the mobs_skills param prevents the TP loss, but then we would need either a separate entry of sheep charge in the global skill list (not this specific mobs list) which we can only do if multiple exist in the dat, or a method of flagging it just for this mob.

According to the dat, these are the skills available to us, with the +256 offset we use in the sql table:
```
(0006)  262 Sheep Charge
(0018)  284 Sheep Charge
(1379) 1635 Sheep Charge
(3603) 3603 Sheep Charge
```
do you know which one is used in the captures of this mob?

Reply:
274 was the param of their auto attack (sheep charge) in the capture from discord

Reply:
It needs a skill ID, not a param. 274 wouldn't be the right skill ID. I'm not sure what's happening there.

Reply:
I can link the timestamp if you'd like to take a look

Reply:
param isn't 274, action ID is. I'll get back to you after I know wtf is happening because 274 is freaking memento mori not sheep charge.  (╯°□°）╯︵ ┻━┻ _grumbles something about SE spaghetti code and wanders off_

Reply:
wondering if packet viewer is adding teh 256 offset we use in our ID lists. if it is that'd put it at 18, which is a legit sheep charge that I typo'd as _284_ above

Reply:
This does prevent TP loss, yes. However they're not gaining any TP from attacking. Which I guess would be expected. Another side effect is it has caused the damage to go from ~40 to ~400